### ISSUE TITLE
Creating Digital Ocean droplets silently fails when hostnames contain "." and "-" characters

### ISSUE THREAD

**Issue by coderigo** [docker/machine#457]
This succeeds:
```bash
docker-machine create -d digitalocean \
        --digitalocean-access-token your1Token2Here \
        --digitalocean-image docker \
        --digitalocean-region sgp1 \
        --digitalocean-size 512mb \
        myhostname
```

This fails:
```bash
docker-machine create -d digitalocean \
        --digitalocean-access-token your1Token2Here \
        --digitalocean-image docker \
        --digitalocean-region sgp1 \
        --digitalocean-size 512mb \
        my.hostname
```

with this error (same behaviour if one uses `my-hostname` instead of `my.hostname`):

```bash
INFO[0000] Creating SSH key...
INFO[0003] Creating Digital Ocean droplet...
INFO[0011] Waiting for SSH...
# Some minutes pass as it waits for an SSH connection.
ERRO[0230] Error creating machine: exit status 255
WARN[0230] You will want to check the provider to make sure the machine and associated resources were properly removed.
FATA[0230] Error creating machine
```

The machine is indeed created (I have to manually nuke it from DigitalOcean) but no SSH connection is ever established.

The error is despite [hosts.go](https://github.com/docker/machine/blob/master/host.go) suggesting in [line 24]() that `.` and `-` are allowed in host names:

```go
validHostNameChars   = `[a-zA-Z0-9\-\.]`
```

I'm no gopher, but sniffing this one out may be a good intro if someone doesn't get to it sooner.

**Comment by ehazlett** [comment]
This is working fine for me:

```
ehazlett ~> docker-machine ls
NAME        ACTIVE   DRIVER         STATE     URL
test-host   *        digitalocean   Running   tcp://104.131.92.53:2376
test.host            digitalocean   Running   tcp://104.236.47.56:2376
ehazlett ~> docker $(docker-machine config test-host) ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
ehazlett ~> docker $(docker-machine config test.host) ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
```

**Comment by thaJeztah** [comment]
Are you both using the same region? With DO, some features are / aren't available, depending on the region / datacenter you're using, wouldn't be surprised if things like this would be different as well.

**Comment by coderigo** [comment]
Thanks @thaJeztah and @ehazlett.
It turns out the region isn't the issue as I tried my command with `sgp1` and `nyc3` and both succeed.

Weirdly enough, this time around trying with `my.hostname`, `my-hostname`, and `myhostname` now works on `sgp1` and `nyc3` regions, and presumably others.

I must have tried at least 5 times before and got the same error but this time it seems to be working without issue. 

I don't want to clog up the issues list with something that can't be reproduced, so I'll close this one, but will come back if I see it again and pay attention to what I did leading up to seeing the issue.

Cheers
